### PR TITLE
Add API to get rankings data for selected spotlight

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game.Tests/Visual/Online/TestSceneRankingsTables.cs [hunk 1]
```diff
@@ -68,6 +68,7 @@ protected override void LoadComplete()
             AddStep("Mania scores", () => createScoreTable(new ManiaRuleset().RulesetInfo));
             AddStep("Taiko country scores", () => createCountryTable(new TaikoRuleset().RulesetInfo));
             AddStep("Catch US performance page 10", () => createPerformanceTable(new CatchRuleset().RulesetInfo, "US", 10));
+            AddStep("Osu 271 spotlight table", () => createSpotlightTable(new OsuRuleset().RulesetInfo, 271));
```
Comment (bdach) [inline osu.Game.Tests/Visual/Online/TestSceneRankingsTables.cs]: Could do with a more descriptive name.

File: osu.Game/Users/UserStatistics.cs [hunk 2]
```diff
@@ -71,21 +71,21 @@ private int rank
...
+            public int? SPlus;
 
             [JsonProperty(@"s")]
             public int S;
 
             [JsonProperty(@"a")]
             public int A;
 
-            public int this[ScoreRank rank]
+            public int? this[ScoreRank rank]
```
Comment (bdach) [inline osu.Game/Users/UserStatistics.cs]: I'm unsure if doing this makes sense if it's going to have to get fixed up to 0 in places anyway (as it is being here in `UserBasedTable`).

Comment (EVAST9919) [reply osu.Game/Users/UserStatistics.cs]: hmm, yeah, we can just make a null check in return

**Review by smoogipoo — APPROVED** [review]
this is fine for now, spotlights are... temporary? as far as I'm hearing

**Pull request merged**